### PR TITLE
Update sql comments upload

### PR DESCRIPTION
**Pull request by jgibson517**
- update sql query for nlpoutput table changes
- set rep_comments equal to all comments if no representative comments are found
- fix generator for pipeline
- fixes bug where num_unique_comments > num_total comments

### CODE REVIEW

File: civiclens/nlp/pipeline.py [hunk 1]
```diff
@@ -76,7 +87,6 @@ def docs_have_titles():
             GROUP BY document_id HAVING COUNT(*) > 20;
             """
     if args.refresh:
```
Comment (abejburton) [inline civiclens/nlp/pipeline.py:89]: Could make sense to have this before `if last_updated is not None:` on line 77. That way if its a refresh you can skip running the longer query, not run that chunk just do the refresh query.

Comment (jgibson517) [reply civiclens/nlp/pipeline.py:95]: nice, good catch!

File: civiclens/nlp/topics.py [hunk 2]
```diff
@@ -207,6 +207,10 @@ def topic_comment_analysis(
         ]
 
     comments += comment_data.to_list()
+    if not comment_data.rep_comments:
+        comment_data.representative = False
+        comments += comment_data.get_nonrepresentative_comments()
```
Comment (abejburton) [inline civiclens/nlp/topics.py:212]: nice, this fixes the bug john pointed out

File: civiclens/nlp/topics.py [hunk 3]
```diff
@@ -218,7 +222,7 @@ def topic_comment_analysis(
         comment.topic_label = topic_labels[comment_topics[comment.id]]
         comment.topic = comment_topics[comment.id]
         comment.sentiment = sentiment_analyzer(comment)
-        if comment.representative:
+        if comment.representative or not comment_data.representative:
```
Comment (abejburton) [inline civiclens/nlp/topics.py:225]: am I missing something? doesn't this say if True or False?

File: civiclens/utils/database_access.py [hunk 4]
```diff
@@ -95,23 +96,26 @@ def upload_comments(connection: Database, comments: RepComments) -> None:
...
             num_representative_comment = EXCLUDED.num_representative_comment,
             topics = EXCLUDED.topics,
-            last_updated = EXCLUDED.last_updated,
+            num_topics = EXCLUDED.num_topics,
+            last_updated = NOW(),
             search_topics = EXCLUDED.search_topics
-        WHERE (regulations_nlpoutput.last_updated < EXCLUDED.last_updated
-                or regulations_nlpoutput.last_updated IS NULL);
+        WHERE regulations_nlpoutput.last_updated IS NULL
+            OR regulations_nlpoutput.last_updated < EXCLUDED.last_updated;
```
Comment (abejburton) [inline civiclens/utils/database_access.py:113]: update statement looks great

File: civiclens/nlp/comments.py [hunk 5]
```diff
@@ -364,9 +364,12 @@ def rep_comment_analysis(id: str, model: SentenceTransformer) -> RepComments:
         comment_data.rep_comments = form_letters + df_rep_paraphrase.to_dicts()
         comment_data.num_representative_comment = len(comment_data.rep_comments)
 
+    num_paraphrased = count_unique_comments(df_paraphrases)
     comment_data.num_total_comments = df.shape[0]
```
Comment (abejburton) [inline civiclens/nlp/comments.py:368]: Do you still have questions on this/are you working through the bug?

One thought I had when you're debugging this: line 150 in `assign_clusters` is a rows variable that equals df.shape[0]. You could return that rows variable and compare it with df.shape[0] in line 368 here to see if there is some kind of accidental transformation that is making the df change size. I'm just trying to think why df_num_paraphrase could possibly have more rows than the df it is a subset of...

File: civiclens/nlp/topics.py (hunk 3 above)
Comment (abejburton) [reply civiclens/nlp/topics.py:240]: just need your thoughts on this, then its good to go

**Review by abejburton — CHANGES_REQUESTED** [review]

**Comment by jgibson517** [comment]
they're different variables! `not comment_data.representative` means there wasn't any represented comments generated in which case we want store all the comments

**Comment by abejburton** [comment]
> they're different variables! `not comment_data.representative` means there wasn't any represented comments generated in which case we want store all the comments

Oh I'm blind lol my bad

**Review by abejburton — APPROVED** [review]

**Pull request merged**